com.hazelcast.client.cache.ClientCacheConfigTest.cacheManagerByLocationClasspathTest

_master_ (commit e3352af34221c58de14ed09dcde0edc9206098c8)

Failed on Oracle JDK 11: http://jenkins.hazelcast.com/view/Official%20Builds/job/Hazelcast-master-OracleJDK11/271/testReport/com.hazelcast.client.cache/ClientCacheConfigTest/cacheManagerByLocationClasspathTest/

Stacktrace:
```
java.lang.AssertionError: expected:<0> but was:<4>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotEquals(Assert.java:835)
	at org.junit.Assert.assertEquals(Assert.java:647)
	at org.junit.Assert.assertEquals(Assert.java:633)
	at com.hazelcast.client.cache.ClientCacheConfigTest.cacheManagerByLocationClasspathTest(ClientCacheConfigTest.java:88)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:115)
	at com.hazelcast.test.FailOnTimeoutStatement$CallableStatement.call(FailOnTimeoutStatement.java:107)
	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
	at java.base/java.lang.Thread.run(Thread.java:834)
```

Standard output:
```
22:26:41,001  INFO |cacheManagerByLocationClasspathTest| - [MetricsConfigHelper] main - [LOCAL] [cluster1] [5.0-SNAPSHOT] Overridden metrics configuration with system property 'hazelcast.metrics.collection.frequency'='1' -> 'MetricsConfig.collectionFrequencySeconds'='1'
22:26:41,003  INFO |cacheManagerByLocationClasspathTest| - [system] main - [127.0.0.1]:5701 [cluster1] [5.0-SNAPSHOT] 
	o  o   O   o---o o--o o      o-o   O    o-o  o-O-o     o--o    o-o  
	|  |  / \     /  |    |     /     / \  |       |       |      o  /o 
	O--O o---o  -O-  O-o  |    O     o---o  o-o    |       o-o    | / | 
	|  | |   |  /    |    |     \    |   |     |   |          |   o/  o 
	o  o o   o o---o o--o O---o  o-o o   o o--o    o       o-o  O  o-o
22:26:41,003  INFO |cacheManagerByLocationClasspathTest| - [system] main - [127.0.0.1]:5701 [cluster1] [5.0-SNAPSHOT] Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
22:26:41,003  INFO |cacheManagerByLocationClasspathTest| - [system] main - [127.0.0.1]:5701 [cluster1] [5.0-SNAPSHOT] Hazelcast Platform 5.0-SNAPSHOT (20210605 - e3352af) starting at [127.0.0.1]:5701
22:26:41,003  INFO |cacheManagerByLocationClasspathTest| - [system] main - [127.0.0.1]:5701 [cluster1] [5.0-SNAPSHOT] Cluster name: cluster1
22:26:41,004  INFO |cacheManagerByLocationClasspathTest| - [MetricsConfigHelper] main - [127.0.0.1]:5701 [cluster1] [5.0-SNAPSHOT] Collecting debug metrics and sending to diagnostics is enabled
22:26:41,011  WARN |cacheManagerByLocationClasspathTest| - [CPSubsystem] main - [127.0.0.1]:5701 [cluster1] [5.0-SNAPSHOT] CP Subsystem is not enabled. CP data structures will operate in UNSAFE mode! Please note that UNSAFE mode will not provide strong consistency guarantees.
22:26:41,016  INFO |cacheManagerByLocationClasspathTest| - [JetServiceBackend] main - [127.0.0.1]:5701 [cluster1] [5.0-SNAPSHOT] Setting number of cooperative threads and default parallelism to 8
22:26:41,017  INFO |cacheManagerByLocationClasspathTest| - [Diagnostics] main - [127.0.0.1]:5701 [cluster1] [5.0-SNAPSHOT] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
22:26:41,018  INFO |cacheManagerByLocationClasspathTest| - [LifecycleService] main - [127.0.0.1]:5701 [cluster1] [5.0-SNAPSHOT] [127.0.0.1]:5701 is STARTING
22:26:41,019  WARN |cacheManagerByLocationClasspathTest| - [Node] main - [127.0.0.1]:5701 [cluster1] [5.0-SNAPSHOT] No join method is enabled! Starting standalone.
22:26:41,019  INFO |cacheManagerByLocationClasspathTest| - [JetExtension] main - [127.0.0.1]:5701 [cluster1] [5.0-SNAPSHOT] Jet extension is enabled after the cluster version upgrade.
22:26:41,019 DEBUG |cacheManagerByLocationClasspathTest| - [JobCoordinationService] hz.fervent_beaver.cached.thread-1 - [127.0.0.1]:5701 [cluster1] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
22:26:41,019  INFO |cacheManagerByLocationClasspathTest| - [JetExtension] main - [127.0.0.1]:5701 [cluster1] [5.0-SNAPSHOT] Jet extension is enabled
22:26:41,019 DEBUG |cacheManagerByLocationClasspathTest| - [JobCoordinationService] hz.fervent_beaver.cached.thread-1 - [127.0.0.1]:5701 [cluster1] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
22:26:41,020  INFO |cacheManagerByLocationClasspathTest| - [LifecycleService] main - [127.0.0.1]:5701 [cluster1] [5.0-SNAPSHOT] [127.0.0.1]:5701 is STARTED
22:26:41,020  INFO |cacheManagerByLocationClasspathTest| - [MetricsConfigHelper] main - [LOCAL] [cluster2] [5.0-SNAPSHOT] Overridden metrics configuration with system property 'hazelcast.metrics.collection.frequency'='1' -> 'MetricsConfig.collectionFrequencySeconds'='1'
22:26:41,021  INFO |cacheManagerByLocationClasspathTest| - [system] main - [127.0.0.1]:5702 [cluster2] [5.0-SNAPSHOT] 
	o  o   O   o---o o--o o      o-o   O    o-o  o-O-o     o--o    o-o  
	|  |  / \     /  |    |     /     / \  |       |       |      o  /o 
	O--O o---o  -O-  O-o  |    O     o---o  o-o    |       o-o    | / | 
	|  | |   |  /    |    |     \    |   |     |   |          |   o/  o 
	o  o o   o o---o o--o O---o  o-o o   o o--o    o       o-o  O  o-o
22:26:41,021  INFO |cacheManagerByLocationClasspathTest| - [system] main - [127.0.0.1]:5702 [cluster2] [5.0-SNAPSHOT] Copyright (c) 2008-2021, Hazelcast, Inc. All Rights Reserved.
22:26:41,021  INFO |cacheManagerByLocationClasspathTest| - [system] main - [127.0.0.1]:5702 [cluster2] [5.0-SNAPSHOT] Hazelcast Platform 5.0-SNAPSHOT (20210605 - e3352af) starting at [127.0.0.1]:5702
22:26:41,021  INFO |cacheManagerByLocationClasspathTest| - [system] main - [127.0.0.1]:5702 [cluster2] [5.0-SNAPSHOT] Cluster name: cluster2
22:26:41,023  INFO |cacheManagerByLocationClasspathTest| - [MetricsConfigHelper] main - [127.0.0.1]:5702 [cluster2] [5.0-SNAPSHOT] Collecting debug metrics and sending to diagnostics is enabled
22:26:41,028  WARN |cacheManagerByLocationClasspathTest| - [CPSubsystem] main - [127.0.0.1]:5702 [cluster2] [5.0-SNAPSHOT] CP Subsystem is not enabled. CP data structures will operate in UNSAFE mode! Please note that UNSAFE mode will not provide strong consistency guarantees.
22:26:41,032  INFO |cacheManagerByLocationClasspathTest| - [JetServiceBackend] main - [127.0.0.1]:5702 [cluster2] [5.0-SNAPSHOT] Setting number of cooperative threads and default parallelism to 8
22:26:41,034  INFO |cacheManagerByLocationClasspathTest| - [Diagnostics] main - [127.0.0.1]:5702 [cluster2] [5.0-SNAPSHOT] Diagnostics disabled. To enable add -Dhazelcast.diagnostics.enabled=true to the JVM arguments.
22:26:41,034  INFO |cacheManagerByLocationClasspathTest| - [LifecycleService] main - [127.0.0.1]:5702 [cluster2] [5.0-SNAPSHOT] [127.0.0.1]:5702 is STARTING
22:26:41,035  WARN |cacheManagerByLocationClasspathTest| - [Node] main - [127.0.0.1]:5702 [cluster2] [5.0-SNAPSHOT] No join method is enabled! Starting standalone.
22:26:41,035  INFO |cacheManagerByLocationClasspathTest| - [JetExtension] main - [127.0.0.1]:5702 [cluster2] [5.0-SNAPSHOT] Jet extension is enabled after the cluster version upgrade.
22:26:41,036 DEBUG |cacheManagerByLocationClasspathTest| - [JobCoordinationService] hz.distracted_beaver.cached.thread-3 - [127.0.0.1]:5702 [cluster2] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
22:26:41,036  INFO |cacheManagerByLocationClasspathTest| - [JetExtension] main - [127.0.0.1]:5702 [cluster2] [5.0-SNAPSHOT] Jet extension is enabled
22:26:41,036 DEBUG |cacheManagerByLocationClasspathTest| - [JobCoordinationService] hz.distracted_beaver.cached.thread-3 - [127.0.0.1]:5702 [cluster2] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
22:26:41,036  INFO |cacheManagerByLocationClasspathTest| - [LifecycleService] main - [127.0.0.1]:5702 [cluster2] [5.0-SNAPSHOT] [127.0.0.1]:5702 is STARTED
22:26:41,183 DEBUG |cacheManagerByLocationClasspathTest| - [JobCoordinationService] hz.fervent_beaver.cached.thread-1 - [127.0.0.1]:5701 [cluster1] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
22:26:41,183 DEBUG |cacheManagerByLocationClasspathTest| - [JobCoordinationService] hz.distracted_beaver.cached.thread-1 - [127.0.0.1]:5702 [cluster2] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
22:26:41,183 DEBUG |cacheManagerByLocationClasspathTest| - [JobCoordinationService] hz.fervent_beaver.cached.thread-3 - [127.0.0.1]:5701 [cluster1] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
22:26:41,183 DEBUG |cacheManagerByLocationClasspathTest| - [JobCoordinationService] hz.distracted_beaver.cached.thread-1 - [127.0.0.1]:5702 [cluster2] [5.0-SNAPSHOT] Not starting jobs because partitions are not yet initialized.
22:26:41,184  WARN |cacheManagerByLocationClasspathTest| - [AuthenticationMessageTask] hz.fervent_beaver.priority-generic-operation.thread-0 - [127.0.0.1]:5701 [cluster1] [5.0-SNAPSHOT] Received auth from Connection[id=1, /127.0.0.1:5701->/127.0.0.1:58748, qualifier=null, endpoint=null, alive=true, connectionType=NONE, planeIndex=-1] with clientUuid ea1abc7e-0178-4072-8fa3-a7cec26eab7b, authentication failed
22:26:41,184  WARN |cacheManagerByLocationClasspathTest| - [AuthenticationMessageTask] hz.distracted_beaver.priority-generic-operation.thread-0 - [127.0.0.1]:5702 [cluster2] [5.0-SNAPSHOT] Received auth from Connection[id=1, /127.0.0.1:5702->/127.0.0.1:42848, qualifier=null, endpoint=null, alive=true, connectionType=NONE, planeIndex=-1] with clientUuid 205e84b0-b83b-41d8-985e-3cb5b80e1499, authentication failed
22:26:41,184  WARN |cacheManagerByLocationClasspathTest| - [AuthenticationMessageTask] hz.fervent_beaver.generic-operation.thread-1 - [127.0.0.1]:5701 [cluster1] [5.0-SNAPSHOT] Received auth from Connection[id=2, /127.0.0.1:5701->/127.0.0.1:53606, qualifier=null, endpoint=null, alive=true, connectionType=NONE, planeIndex=-1] with clientUuid 205e84b0-b83b-41d8-985e-3cb5b80e1499, authentication failed
22:26:41,184  WARN |cacheManagerByLocationClasspathTest| - [AuthenticationMessageTask] hz.distracted_beaver.generic-operation.thread-0 - [127.0.0.1]:5702 [cluster2] [5.0-SNAPSHOT] Received auth from Connection[id=2, /127.0.0.1:5702->/127.0.0.1:47837, qualifier=null, endpoint=null, alive=true, connectionType=NONE, planeIndex=-1] with clientUuid ea1abc7e-0178-4072-8fa3-a7cec26eab7b, authentication failed
22:26:41,184  WARN || - [TcpClientConnection] hz.client_180.internal-8 - hz.client_180 [8c360673-7e50-465a-bfab-62d91d5ef3ac] [5.0-SNAPSHOT] [5.0-SNAPSHOT] ClientConnection{alive=false, connectionId=21, channel=NioChannel{/127.0.0.1:47837->/127.0.0.1:5702}, remoteAddress=null, lastReadTime=2021-06-05 22:26:41.184, lastWriteTime=2021-06-05 22:26:41.184, closedTime=2021-06-05 22:26:41.184, connected server version=null} closed. Reason: Failed to authenticate connection
com.hazelcast.client.AuthenticationException: Authentication failed. The configured cluster name on the client (see ClientConfig.setClusterName()) does not match the one configured in the cluster or the credentials set in the Client security config could not be authenticated
	at com.hazelcast.client.impl.connection.tcp.TcpClientConnectionManager.checkAuthenticationResponse(TcpClientConnectionManager.java:982) ~[classes/:?]
	at com.hazelcast.client.impl.connection.tcp.TcpClientConnectionManager.onAuthenticated(TcpClientConnectionManager.java:885) ~[classes/:?]
	at com.hazelcast.client.impl.connection.tcp.TcpClientConnectionManager.getOrConnectToMember(TcpClientConnectionManager.java:622) ~[classes/:?]
	at com.hazelcast.client.impl.connection.tcp.TcpClientConnectionManager$ConnectToAllClusterMembersTask.lambda$run$0(TcpClientConnectionManager.java:1169) ~[classes/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:76) [classes/:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102) [classes/:?]
22:26:41,184  WARN || - [TcpClientConnection] hz.client_181.internal-8 - hz.client_181 [8c360673-7e50-465a-bfab-62d91d5ef3ac] [5.0-SNAPSHOT] [5.0-SNAPSHOT] ClientConnection{alive=false, connectionId=21, channel=NioChannel{/127.0.0.1:42848->/127.0.0.1:5702}, remoteAddress=null, lastReadTime=2021-06-05 22:26:41.184, lastWriteTime=2021-06-05 22:26:41.184, closedTime=2021-06-05 22:26:41.184, connected server version=null} closed. Reason: Failed to authenticate connection
com.hazelcast.client.AuthenticationException: Authentication failed. The configured cluster name on the client (see ClientConfig.setClusterName()) does not match the one configured in the cluster or the credentials set in the Client security config could not be authenticated
	at com.hazelcast.client.impl.connection.tcp.TcpClientConnectionManager.checkAuthenticationResponse(TcpClientConnectionManager.java:982) ~[classes/:?]
	at com.hazelcast.client.impl.connection.tcp.TcpClientConnectionManager.onAuthenticated(TcpClientConnectionManager.java:885) ~[classes/:?]
	at com.hazelcast.client.impl.connection.tcp.TcpClientConnectionManager.getOrConnectToMember(TcpClientConnectionManager.java:622) ~[classes/:?]
	at com.hazelcast.client.impl.connection.tcp.TcpClientConnectionManager$ConnectToAllClusterMembersTask.lambda$run$0(TcpClientConnectionManager.java:1169) ~[classes/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:76) [classes/:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102) [classes/:?]
22:26:41,184  INFO |cacheManagerByLocationClasspathTest| - [TcpServerConnection] hz.distracted_beaver.IO.thread-in-1 - [127.0.0.1]:5702 [cluster2] [5.0-SNAPSHOT] Connection[id=2, /127.0.0.1:5702->/127.0.0.1:47837, qualifier=null, endpoint=null, alive=false, connectionType=NONE, planeIndex=-1] closed. Reason: Connection closed by the other side
22:26:41,184  INFO |cacheManagerByLocationClasspathTest| - [TcpServerConnection] hz.distracted_beaver.IO.thread-in-0 - [127.0.0.1]:5702 [cluster2] [5.0-SNAPSHOT] Connection[id=1, /127.0.0.1:5702->/127.0.0.1:42848, qualifier=null, endpoint=null, alive=false, connectionType=NONE, planeIndex=-1] closed. Reason: Connection closed by the other side
22:26:41,184  WARN || - [ClientConnectionManager] hz.client_181.internal-8 - hz.client_181 [8c360673-7e50-465a-bfab-62d91d5ef3ac] [5.0-SNAPSHOT] [5.0-SNAPSHOT] Could not connect to member 61f5f80f-5b1a-4df6-9513-81e625500414, reason com.hazelcast.client.AuthenticationException: Authentication failed. The configured cluster name on the client (see ClientConfig.setClusterName()) does not match the one configured in the cluster or the credentials set in the Client security config could not be authenticated
22:26:41,184  WARN || - [ClientConnectionManager] hz.client_180.internal-8 - hz.client_180 [8c360673-7e50-465a-bfab-62d91d5ef3ac] [5.0-SNAPSHOT] [5.0-SNAPSHOT] Could not connect to member 61f5f80f-5b1a-4df6-9513-81e625500414, reason com.hazelcast.client.AuthenticationException: Authentication failed. The configured cluster name on the client (see ClientConfig.setClusterName()) does not match the one configured in the cluster or the credentials set in the Client security config could not be authenticated
22:26:41,184  WARN || - [TcpClientConnection] hz.client_181.internal-6 - hz.client_181 [8c360673-7e50-465a-bfab-62d91d5ef3ac] [5.0-SNAPSHOT] [5.0-SNAPSHOT] ClientConnection{alive=false, connectionId=22, channel=NioChannel{/127.0.0.1:53606->/127.0.0.1:5701}, remoteAddress=null, lastReadTime=2021-06-05 22:26:41.184, lastWriteTime=2021-06-05 22:26:41.184, closedTime=2021-06-05 22:26:41.184, connected server version=null} closed. Reason: Failed to authenticate connection
com.hazelcast.client.AuthenticationException: Authentication failed. The configured cluster name on the client (see ClientConfig.setClusterName()) does not match the one configured in the cluster or the credentials set in the Client security config could not be authenticated
	at com.hazelcast.client.impl.connection.tcp.TcpClientConnectionManager.checkAuthenticationResponse(TcpClientConnectionManager.java:982) ~[classes/:?]
	at com.hazelcast.client.impl.connection.tcp.TcpClientConnectionManager.onAuthenticated(TcpClientConnectionManager.java:885) ~[classes/:?]
	at com.hazelcast.client.impl.connection.tcp.TcpClientConnectionManager.getOrConnectToMember(TcpClientConnectionManager.java:622) ~[classes/:?]
	at com.hazelcast.client.impl.connection.tcp.TcpClientConnectionManager$ConnectToAllClusterMembersTask.lambda$run$0(TcpClientConnectionManager.java:1169) ~[classes/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:76) [classes/:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102) [classes/:?]
22:26:41,185  WARN || - [TcpClientConnection] hz.client_180.internal-10 - hz.client_180 [8c360673-7e50-465a-bfab-62d91d5ef3ac] [5.0-SNAPSHOT] [5.0-SNAPSHOT] ClientConnection{alive=false, connectionId=22, channel=NioChannel{/127.0.0.1:58748->/127.0.0.1:5701}, remoteAddress=null, lastReadTime=2021-06-05 22:26:41.184, lastWriteTime=2021-06-05 22:26:41.184, closedTime=2021-06-05 22:26:41.184, connected server version=null} closed. Reason: Failed to authenticate connection
com.hazelcast.client.AuthenticationException: Authentication failed. The configured cluster name on the client (see ClientConfig.setClusterName()) does not match the one configured in the cluster or the credentials set in the Client security config could not be authenticated
	at com.hazelcast.client.impl.connection.tcp.TcpClientConnectionManager.checkAuthenticationResponse(TcpClientConnectionManager.java:982) ~[classes/:?]
	at com.hazelcast.client.impl.connection.tcp.TcpClientConnectionManager.onAuthenticated(TcpClientConnectionManager.java:885) ~[classes/:?]
	at com.hazelcast.client.impl.connection.tcp.TcpClientConnectionManager.getOrConnectToMember(TcpClientConnectionManager.java:622) ~[classes/:?]
	at com.hazelcast.client.impl.connection.tcp.TcpClientConnectionManager$ConnectToAllClusterMembersTask.lambda$run$0(TcpClientConnectionManager.java:1169) ~[classes/:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) [?:?]
	at java.util.concurrent.FutureTask.run(FutureTask.java:264) [?:?]
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304) [?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) [?:?]
	at java.lang.Thread.run(Thread.java:834) [?:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.executeRun(HazelcastManagedThread.java:76) [classes/:?]
	at com.hazelcast.internal.util.executor.HazelcastManagedThread.run(HazelcastManagedThread.java:102) [classes/:?]
22:26:41,185  WARN || - [ClientConnectionManager] hz.client_181.internal-6 - hz.client_181 [8c360673-7e50-465a-bfab-62d91d5ef3ac] [5.0-SNAPSHOT] [5.0-SNAPSHOT] Could not connect to member a1df6606-8e64-4ddb-94b9-055670dd9871, reason com.hazelcast.client.AuthenticationException: Authentication failed. The configured cluster name on the client (see ClientConfig.setClusterName()) does not match the one configured in the cluster or the credentials set in the Client security config could not be authenticated
22:26:41,185  INFO |cacheManagerByLocationClasspathTest| - [TcpServerConnection] hz.fervent_beaver.IO.thread-in-0 - [127.0.0.1]:5701 [cluster1] [5.0-SNAPSHOT] Connection[id=1, /127.0.0.1:5701->/127.0.0.1:58748, qualifier=null, endpoint=null, alive=false, connectionType=NONE, planeIndex=-1] closed. Reason: Connection closed by the other side
22:26:41,185  WARN || - [ClientConnectionManager] hz.client_180.internal-10 - hz.client_180 [8c360673-7e50-465a-bfab-62d91d5ef3ac] [5.0-SNAPSHOT] [5.0-SNAPSHOT] Could not connect to member a1df6606-8e64-4ddb-94b9-055670dd9871, reason com.hazelcast.client.AuthenticationException: Authentication failed. The configured cluster name on the client (see ClientConfig.setClusterName()) does not match the one configured in the cluster or the credentials set in the Client security config could not be authenticated
22:26:41,185  INFO |cacheManagerByLocationClasspathTest| - [TcpServerConnection] hz.fervent_beaver.IO.thread-in-1 - [127.0.0.1]:5701 [cluster1] [5.0-SNAPSHOT] Connection[id=2, /127.0.0.1:5701->/127.0.0.1:53606, qualifier=null, endpoint=null, alive=false, connectionType=NONE, planeIndex=-1] closed. Reason: Connection closed by the other side
22:26:41,219  INFO |cacheManagerByLocationClasspathTest| - [LifecycleService] main - hz.client_180 [8c360673-7e50-465a-bfab-62d91d5ef3ac] [5.0-SNAPSHOT] [5.0-SNAPSHOT] HazelcastClient 5.0-SNAPSHOT (20210605 - e3352af) is SHUTTING_DOWN
22:26:41,220  INFO || - [ClientConnectionManager] hz.client_180.internal-7 - hz.client_180 [8c360673-7e50-465a-bfab-62d91d5ef3ac] [5.0-SNAPSHOT] [5.0-SNAPSHOT] Unable to connect to any address from the cluster with name: 8c360673-7e50-465a-bfab-62d91d5ef3ac. The following addresses were tried: [[127.0.0.1]:5702, [127.0.0.1]:5701]
22:26:41,220  WARN || - [ClientConnectionManager] hz.client_180.internal-7 - hz.client_180 [8c360673-7e50-465a-bfab-62d91d5ef3ac] [5.0-SNAPSHOT] [5.0-SNAPSHOT] Could not connect to any cluster, shutting down the client: Client is being shutdown.
22:26:41,221  INFO |cacheManagerByLocationClasspathTest| - [LifecycleService] main - hz.client_180 [8c360673-7e50-465a-bfab-62d91d5ef3ac] [5.0-SNAPSHOT] [5.0-SNAPSHOT] HazelcastClient 5.0-SNAPSHOT (20210605 - e3352af) is SHUTDOWN
22:26:41,221  INFO |cacheManagerByLocationClasspathTest| - [LifecycleService] main - hz.client_207 [8c360673-7e50-465a-bfab-62d91d5ef3ac] [5.0-SNAPSHOT] [5.0-SNAPSHOT] HazelcastClient 5.0-SNAPSHOT (20210605 - e3352af) is SHUTTING_DOWN
22:26:41,221  INFO || - [ClientConnectionManager] hz.client_207.internal-1 - hz.client_207 [8c360673-7e50-465a-bfab-62d91d5ef3ac] [5.0-SNAPSHOT] [5.0-SNAPSHOT] Unable to connect to any address from the cluster with name: 8c360673-7e50-465a-bfab-62d91d5ef3ac. The following addresses were tried: [[127.0.0.1]:5702, [127.0.0.1]:5701]
22:26:41,221  WARN || - [ClientConnectionManager] hz.client_207.internal-1 - hz.client_207 [8c360673-7e50-465a-bfab-62d91d5ef3ac] [5.0-SNAPSHOT] [5.0-SNAPSHOT] Could not connect to any cluster, shutting down the client: Client is being shutdown.
22:26:41,228  INFO |cacheManagerByLocationClasspathTest| - [LifecycleService] main - hz.client_207 [8c360673-7e50-465a-bfab-62d91d5ef3ac] [5.0-SNAPSHOT] [5.0-SNAPSHOT] HazelcastClient 5.0-SNAPSHOT (20210605 - e3352af) is SHUTDOWN
22:26:41,228  INFO |cacheManagerByLocationClasspathTest| - [LifecycleService] main - hz.client_206 [8c360673-7e50-465a-bfab-62d91d5ef3ac] [5.0-SNAPSHOT] [5.0-SNAPSHOT] HazelcastClient 5.0-SNAPSHOT (20210605 - e3352af) is SHUTTING_DOWN
22:26:41,228  INFO || - [ClientConnectionManager] hz.client_206.internal-1 - hz.client_206 [8c360673-7e50-465a-bfab-62d91d5ef3ac] [5.0-SNAPSHOT] [5.0-SNAPSHOT] Unable to connect to any address from the cluster with name: 8c360673-7e50-465a-bfab-62d91d5ef3ac. The following addresses were tried: [[127.0.0.1]:5702, [127.0.0.1]:5701]
22:26:41,228  WARN || - [ClientConnectionManager] hz.client_206.internal-1 - hz.client_206 [8c360673-7e50-465a-bfab-62d91d5ef3ac] [5.0-SNAPSHOT] [5.0-SNAPSHOT] Could not connect to any cluster, shutting down the client: Client is being shutdown.
22:26:41,229  INFO |cacheManagerByLocationClasspathTest| - [LifecycleService] main - hz.client_206 [8c360673-7e50-465a-bfab-62d91d5ef3ac] [5.0-SNAPSHOT] [5.0-SNAPSHOT] HazelcastClient 5.0-SNAPSHOT (20210605 - e3352af) is SHUTDOWN
22:26:41,230  INFO |cacheManagerByLocationClasspathTest| - [LifecycleService] main - hz.client_181 [8c360673-7e50-465a-bfab-62d91d5ef3ac] [5.0-SNAPSHOT] [5.0-SNAPSHOT] HazelcastClient 5.0-SNAPSHOT (20210605 - e3352af) is SHUTTING_DOWN
22:26:41,230  INFO || - [ClientConnectionManager] hz.client_181.internal-7 - hz.client_181 [8c360673-7e50-465a-bfab-62d91d5ef3ac] [5.0-SNAPSHOT] [5.0-SNAPSHOT] Unable to connect to any address from the cluster with name: 8c360673-7e50-465a-bfab-62d91d5ef3ac. The following addresses were tried: [[127.0.0.1]:5702, [127.0.0.1]:5701]
22:26:41,230  WARN || - [ClientConnectionManager] hz.client_181.internal-7 - hz.client_181 [8c360673-7e50-465a-bfab-62d91d5ef3ac] [5.0-SNAPSHOT] [5.0-SNAPSHOT] Could not connect to any cluster, shutting down the client: Client is being shutdown.
22:26:41,231  INFO |cacheManagerByLocationClasspathTest| - [LifecycleService] main - hz.client_181 [8c360673-7e50-465a-bfab-62d91d5ef3ac] [5.0-SNAPSHOT] [5.0-SNAPSHOT] HazelcastClient 5.0-SNAPSHOT (20210605 - e3352af) is SHUTDOWN
22:26:41,232  INFO |cacheManagerByLocationClasspathTest| - [LifecycleService] main - [127.0.0.1]:5702 [cluster2] [5.0-SNAPSHOT] [127.0.0.1]:5702 is SHUTTING_DOWN
22:26:41,232 DEBUG |cacheManagerByLocationClasspathTest| - [JobCoordinationService] main - [127.0.0.1]:5702 [cluster2] [5.0-SNAPSHOT] Added a shutting-down member: 80e77843-064b-4fd8-9447-65f147ee9c73
22:26:41,232  INFO |cacheManagerByLocationClasspathTest| - [Node] main - [127.0.0.1]:5702 [cluster2] [5.0-SNAPSHOT] Shutting down connection manager...
22:26:41,234  INFO |cacheManagerByLocationClasspathTest| - [Node] main - [127.0.0.1]:5702 [cluster2] [5.0-SNAPSHOT] Shutting down node engine...
22:26:41,235  INFO |cacheManagerByLocationClasspathTest| - [NodeExtension] main - [127.0.0.1]:5702 [cluster2] [5.0-SNAPSHOT] Destroying node NodeExtension.
22:26:41,235  INFO |cacheManagerByLocationClasspathTest| - [Node] main - [127.0.0.1]:5702 [cluster2] [5.0-SNAPSHOT] Hazelcast Shutdown is completed in 3 ms.
22:26:41,235  INFO |cacheManagerByLocationClasspathTest| - [LifecycleService] main - [127.0.0.1]:5702 [cluster2] [5.0-SNAPSHOT] [127.0.0.1]:5702 is SHUTDOWN
22:26:41,235  INFO |cacheManagerByLocationClasspathTest| - [LifecycleService] main - [127.0.0.1]:5701 [cluster1] [5.0-SNAPSHOT] [127.0.0.1]:5701 is SHUTTING_DOWN
22:26:41,235 DEBUG |cacheManagerByLocationClasspathTest| - [JobCoordinationService] main - [127.0.0.1]:5701 [cluster1] [5.0-SNAPSHOT] Added a shutting-down member: 05f08494-752f-4350-86f4-3af4412c0120
22:26:41,236  INFO |cacheManagerByLocationClasspathTest| - [Node] main - [127.0.0.1]:5701 [cluster1] [5.0-SNAPSHOT] Shutting down connection manager...
22:26:41,236  INFO |cacheManagerByLocationClasspathTest| - [Node] main - [127.0.0.1]:5701 [cluster1] [5.0-SNAPSHOT] Shutting down node engine...
22:26:41,238  INFO |cacheManagerByLocationClasspathTest| - [NodeExtension] main - [127.0.0.1]:5701 [cluster1] [5.0-SNAPSHOT] Destroying node NodeExtension.
22:26:41,238  INFO |cacheManagerByLocationClasspathTest| - [Node] main - [127.0.0.1]:5701 [cluster1] [5.0-SNAPSHOT] Hazelcast Shutdown is completed in 3 ms.
22:26:41,238  INFO |cacheManagerByLocationClasspathTest| - [LifecycleService] main - [127.0.0.1]:5701 [cluster1] [5.0-SNAPSHOT] [127.0.0.1]:5701 is SHUTDOWN
BuildInfo right after cacheManagerByLocationClasspathTest(com.hazelcast.client.cache.ClientCacheConfigTest): BuildInfo{version='5.0-SNAPSHOT', build='20210605', buildNumber=20210605, revision=e3352af, enterprise=false, serializationVersion=1, jet=JetBuildInfo{version='5.0-SNAPSHOT', build='20210605', revision='e3352af'}}
Hiccups measured while running test 'cacheManagerByLocationClasspathTest(com.hazelcast.client.cache.ClientCacheConfigTest):'
22:26:40, accumulated pauses: 181 ms, max pause: 146 ms, pauses over 1000 ms: 0
```

--- a/hazelcast/src/main/java/com/hazelcast/jet/pipeline/Sinks.java
+++ b/hazelcast/src/main/java/com/hazelcast/jet/pipeline/Sinks.java
@@ -18,12 +18,12 @@ package com.hazelcast.jet.pipeline;
 
 import com.hazelcast.client.config.ClientConfig;
 import com.hazelcast.collection.IList;
+import com.hazelcast.core.HazelcastInstance;
 import com.hazelcast.core.Offloadable;
 import com.hazelcast.function.BiConsumerEx;
 import com.hazelcast.function.BiFunctionEx;
 import com.hazelcast.function.BinaryOperatorEx;
 import com.hazelcast.function.FunctionEx;
-import com.hazelcast.security.impl.function.SecuredFunctions;
 import com.hazelcast.function.SupplierEx;
 import com.hazelcast.jet.JetService;
 import com.hazelcast.jet.Observable;
@@ -34,6 +34,7 @@ import com.hazelcast.jet.impl.pipeline.SinkImpl;
 import com.hazelcast.jet.json.JsonUtil;
 import com.hazelcast.map.EntryProcessor;
 import com.hazelcast.map.IMap;
+import com.hazelcast.security.impl.function.SecuredFunctions;
 import com.hazelcast.security.permission.ReliableTopicPermission;
 import com.hazelcast.topic.ITopic;
 
@@ -64,6 +65,7 @@ import static com.hazelcast.jet.core.processor.SinkProcessors.writeRemoteCacheP;
 import static com.hazelcast.jet.core.processor.SinkProcessors.writeRemoteListP;
 import static com.hazelcast.jet.core.processor.SinkProcessors.writeRemoteMapP;
 import static com.hazelcast.jet.core.processor.SinkProcessors.writeSocketP;
+import static com.hazelcast.jet.datamodel.Tuple2.tuple2;
 import static com.hazelcast.jet.impl.util.ImdgUtil.asClientConfig;
 import static com.hazelcast.jet.impl.util.ImdgUtil.asXmlString;
 import static com.hazelcast.security.permission.ActionConstants.ACTION_CREATE;
@@ -889,9 +891,15 @@ public final class Sinks {
     @Nonnull
     public static <T> Sink<T> remoteReliableTopic(@Nonnull String reliableTopicName, @Nonnull ClientConfig clientConfig) {
         String clientXml = asXmlString(clientConfig); //conversion needed for serializability
-        return SinkBuilder.<ITopic<T>>sinkBuilder("reliableTopicSink(" + reliableTopicName + "))",
-                ctx -> newHazelcastClient(asClientConfig(clientXml)).getReliableTopic(reliableTopicName))
-                .<T>receiveFn(ITopic::publish)
+        return SinkBuilder
+                .sinkBuilder("reliableTopicSink(" + reliableTopicName + "))",
+                        ctx -> {
+                            HazelcastInstance client = newHazelcastClient(asClientConfig(clientXml));
+                            ITopic<T> topic = client.getReliableTopic(reliableTopicName);
+                            return tuple2(client, topic);
+                        })
+                .<T>receiveFn((clientTopicTuple, message) -> clientTopicTuple.f1().publish(message))
+                .destroyFn(clientTopicTuple -> clientTopicTuple.f0().shutdown())
                 .build();
     }
 

--- a/hazelcast/src/test/java/com/hazelcast/jet/pipeline/SinksTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/jet/pipeline/SinksTest.java
@@ -49,7 +49,6 @@ import com.hazelcast.nio.serialization.DataSerializable;
 import com.hazelcast.partition.strategy.StringPartitioningStrategy;
 import com.hazelcast.test.annotation.QuickTest;
 import com.hazelcast.test.annotation.SlowTest;
-import org.junit.After;
 import org.junit.AfterClass;
 import org.junit.BeforeClass;
 import org.junit.Test;

--- a/hazelcast/src/test/java/com/hazelcast/jet/pipeline/SinksTest.java
+++ b/hazelcast/src/test/java/com/hazelcast/jet/pipeline/SinksTest.java
@@ -17,6 +17,7 @@
 package com.hazelcast.jet.pipeline;
 
 import com.hazelcast.cache.ICache;
+import com.hazelcast.client.HazelcastClient;
 import com.hazelcast.client.config.ClientConfig;
 import com.hazelcast.config.CacheSimpleConfig;
 import com.hazelcast.config.Config;
@@ -48,6 +49,7 @@ import com.hazelcast.nio.serialization.DataSerializable;
 import com.hazelcast.partition.strategy.StringPartitioningStrategy;
 import com.hazelcast.test.annotation.QuickTest;
 import com.hazelcast.test.annotation.SlowTest;
+import org.junit.After;
 import org.junit.AfterClass;
 import org.junit.BeforeClass;
 import org.junit.Test;
@@ -69,6 +71,7 @@ import static com.hazelcast.jet.impl.pipeline.AbstractStage.transformOf;
 import static com.hazelcast.jet.json.JsonUtil.hazelcastJsonValue;
 import static java.util.stream.Collectors.toList;
 import static java.util.stream.Collectors.toMap;
+import static org.assertj.core.api.Assertions.assertThat;
 import static org.junit.Assert.assertEquals;
 import static org.junit.Assert.assertTrue;
 
@@ -913,6 +916,19 @@ public class SinksTest extends PipelineTestSupport {
 
         // Then
         assertTrueEventually(() -> assertEquals(itemCount, receivedList.size()));
+    }
+
+    @Test
+    public void remoteReliableTopicSinkClosesClient() {
+        // Check we are in a clean state
+        assertThat(HazelcastClient.getAllHazelcastClients()).hasSize(0);
+
+        // When
+        Sink<Object> sink = Sinks.remoteReliableTopic(sinkName, clientConfig);
+        p.readFrom(Sources.list(srcName)).writeTo(sink);
+        execute();
+
+        assertThat(HazelcastClient.getAllHazelcastClients()).hasSize(0);
     }
 
     @Test

